src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,111 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node<K, V>[] table;
+    private final double loadFactor;
+    private int size;
+    private int capacity;
+    private final int defaultInitialCapacity = 1 << 4;
+    private final double defaultLoadFactor = 0.75;
+
+    public MyHashMap() {
+        this.loadFactor = defaultLoadFactor;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        if (table == null) {
+            resize();
+        }
+        int pos = hash % capacity;
+        if (table[pos] == null) {
+            table[pos] = new Node<>(key, value, hash, null);
+            sizePlus();
+        } else if (table[pos].hash == hash && (Objects.equals(key, table[pos].key))) {
+            table[pos].value = value;
+        } else {
+            Node<K, V> tempX;
+            Node<K, V> tempY = table[pos];
+            while (true) {
+                if ((tempX = tempY.next) == null) {
+                    tempY.next = new Node<>(key, value, hash, null);
+                    sizePlus();
+                    break;
+                }
+                if (tempX.hash == hash && (Objects.equals(key, tempX.key))) {
+                    tempX.value = value;
+                    break;
+                }
+                tempY = tempX;
+            }
```

Comment:
tempX, tempY = bad naming, and we need only one variable which equals table[pos]. `while(true)` is bad practice. Logic can be simpler: 
![image](https://user-images.githubusercontent.com/49767049/214278536-6e709f47-f159-4946-9f1f-27e1252f9425.png)